prime-router/docs/proposals/0017-metadata-tables-cleanup.md

```diff
@@ -0,0 +1,441 @@
+    , sending_application_name          varchar(128) null
+    , sending_application_id            varchar(128) null
+    , created_at                        timestamp not null default current_timestamp
+    -- patient info
+    , patient_state_id                  bigint null references state(state_id)
+    , patient_county_id                 bigint null references county(county_id)
+    , patient_country_id                bigint null references country(country_id)
+    , patient_postal_code               varchar(64) null
+    , patient_gender_code               varchar(64) null
+    , patient_gender                    varchar(128) null
+    , patient_ethnicity_code            varchar(64) null
+    , patient_ethnicity                 varchar(128) null
+    , patient_race_code                 varchar(64) null
+    , patient_race                      varchar(128) null
+    , patient_tribal_citizenship_code   varchar(64) null
+    , patient_tribal_citizenship        varchar(128) null
+    , patient_species_code              varchar(64) null
+    , patient_species                   varchar(128) null
+    , patient_age                       varchar(64) null
+);
+
+comment on table elr_message is
+    'Our main ELR message table. This represents a single ORU_R01 message, though the data does not need to necessarily come from HL7.';
+comment on column elr_message.elr_message_id is
+    'A sequential primary key for the message table';
+
+
+-- there can be many AOE questions for a test. let's catch them here
+create table elr_message_aoe_question
+(
+    elr_message_id                      bigint not null
+    , aoe_set_id                        bigint not null default 1
+    , aoe_question_id                   varchar(64) not null
+    , aoe_question_text                 varchar(512) not null
+    , aoe_result_code                   varchar(64) not null
+    , aoe_result                        varchar(512) not null
+);
+
+-- add a composite key
+alter table elr_message_aoe_question
+    add constraint elr_message_aoe_question_pk
+        primary key (elr_message_id, aoe_set_id, aoe_question_id)
+;
+
+
+-- our collection of requested tests. there can be multiple tests per ELR message
+create table elr_message_ordered_test
+(
+    elr_message_ordered_test_id         bigserial primary key
+    , elr_message_id                    bigint not null references elr_message(elr_message_id)
+    , test_order_id                     varchar(512) not null
+    , test_ordered_code                 varchar(64) not null
+    , test_ordered_long_name            varchar(1024) not null
+    , test_ordered_encoding_version     varchar(64) null
+    , test_set_id                       int not null default 1
+    , reason_for_study_code             varchar(64) null
+    , reason_for_study                  varchar(512) null
+    -- TODO: should we split out the test results to a distinct table? can a test have more than one result?
```

Comment:
I don't know that we split out results to a distinct table unless it becomes necessary for performance. A test can have definitely have more than one result for the same message id. ex: there are many resp viral panels that test both flu and sars/Covid. where you would have one order (test ordered loinc 95380-2) but four results (test performed LOINCs -94500-6, 94502-2, 92142-9, 92141-1)